src/libstd/sys/common/net2.rs

```diff
@@ -127,6 +128,50 @@ pub fn lookup_host(host: &str) -> io::Result<LookupHost> {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+// lookup_addr
+////////////////////////////////////////////////////////////////////////////////
+
+extern "system" {
+    fn getnameinfo(sa: *const libc::sockaddr, salen: socklen_t,
+        host: *mut c_char, hostlen: libc::size_t,
+        serv: *mut c_char, servlen: libc::size_t,
+        flags: c_int) -> c_int;
+}
+
+const NI_MAXHOST: usize = 1025;
+
+pub fn lookup_addr(addr: IpAddr) -> io::Result<String> {
+    init();
+
+    let addr = SocketAddr::new(addr, 0);
+    let (inner, len) = addr.into_inner();
+    let mut hostbuf = [0 as c_char; NI_MAXHOST];
+
+    let data = unsafe {
+        try!(cvt_gai(getnameinfo(inner, len,
+            hostbuf.as_mut_ptr(), NI_MAXHOST as libc::size_t,
+            0 as *mut _, 0, 0)));
+
+        CStr::from_ptr(hostbuf.as_ptr())
+    };
+
+    let name = from_utf8(data.to_bytes()).unwrap();
+
+    // `getnameinfo` returns the input IP address as a string when the name has
+    // failed to be resolved. The `NI_NAMEREQD` flag will override this behavior,
```

Comment:
I don't particularly mind either way on whether this is actually an error or not, but right now it's pretty safe to add the constant assuming GNU libc. There are likely *many* compatibility problems with the standard library and other libc implementations (largely because of all the definitions in the `liblibc` crate), so we can cross that bridge of updating all the constants when we get there.